### DISCUSSION TITLE
How to indicate Size has changed from custom control?

### DISCUSSION THREAD

**Discussion by MaxwellDAssistek** [AvaloniaUI/Avalonia#15854] [Q&A]
Hello,

I am making a custom native control using `NativeControlHost`. I have overridden `MeasureOverride` to properly return the actual size of the native control. 

The problem is that I'm not sure how to get Avalonia to re-measure the layout if the size of the native control changed. I tried calling `UpdateLayout()`, but I see that `MeasureOverride` is not getting called anymore.

Maxwell.

**Comment by stevemonaco** [comment]
You should try calling `InvalidateMeasure` on the custom control. This enqueues the control to be measured when the `UpdateLayout` evaluates a layout pass. `UpdateLayout` should be called automatically by Avalonia anyways when there are invalid items (measure or arrange).

**Reply by stevemonaco** [reply]
You can also use `AffectsMeasure` to wire dependency properties to automatically do this. eg. https://github.com/AvaloniaUI/Avalonia/blob/861c0e4c42dc1f28a50ad0f516c64823618093f6/src/Avalonia.Controls/Image.cs#L35-L40
